packages/api/src/module/write/email-confirmation/domain/approve-email-confirmation.ts

```diff
@@ -0,0 +1,20 @@
+import { ApproveEmailConfirmation } from '@/module/commands/approve-email-confirmation';
+import { emailConfirmationWasApprovedEvent } from '@/module/events/email-confirmation-was-approved.domain.event';
+
+import { EmailConfirmationDomainEvent } from './events';
+
+export const approveEmailConfirmation =
+  (command: ApproveEmailConfirmation) =>
+  (pastEvents: EmailConfirmationDomainEvent[]): EmailConfirmationDomainEvent[] => {
+    const lastPublishedEmailConfirmation = pastEvents.pop();
+
+    if (!lastPublishedEmailConfirmation) throw new Error("Could't find request which could be approved");
+
+    if (lastPublishedEmailConfirmation.type === 'EmailConfirmationWasApproved')
+      throw new Error('Email has been already confirmed');
```

Comment:
Let's change to `Email confirmation has been already approved`
Because it's not confirmation of email, but confirmation of certain action (confirmationFor) by mean of email.